src/VisualStudio/Core/Def/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -221,6 +282,45 @@ public int FormatSpan(IVsTextLines pBuffer, VsTextSpan[] tsInSurfaceBuffer)
 
             var formattingSpan = CommonFormattingHelpers.GetFormattingSpan(SubjectBuffer.CurrentSnapshot, snippetTrackingSpan.GetSpan(SubjectBuffer.CurrentSnapshot));
 
+            // If the snippet spans more than one line and a specific line ending is expected, normalize the line
+            // endings inside the snippet span before continuing.
+            if (newLineEnding is not null && tsInSurfaceBuffer[0].iEndLine > tsInSurfaceBuffer[0].iStartLine)
```

Comment:
📝 This block forces normalization of newlines within the inserted snippet. It is the fix for #67739.